lib/Sema/CSBindings.cpp

```diff
@@ -1629,11 +1667,30 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("delayed");
   if (isSubtypeOfExistentialType())
     attributes.push_back("subtype_of_existential");
-  auto literalKind = getLiteralKind();
+  auto literalKind = getLiteralForScore();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
-    attributes.push_back(literalAttrStr);
+    std::vector<std::string> kinds;
+    std::string literalAttrStr;
+    literalAttrStr.append("[literal: ");
+    if (literalKind == LiteralBindingKind::Atom) {
+      if (auto atomKind = TypeVar->getImpl().getAtomicLiteralKind()) {
+        kinds.push_back(getAtomLiteral(*atomKind).str());
+      }
+    } else if (literalKind == LiteralBindingKind::Collection) {
+      for (const auto &protocolKind : getLiteralRequirements()) {
+        kinds.push_back((getCollectionLiteral(protocolKind) + " " +
+                         getLiteralBindingKind(literalKind))
```

Comment:
Do we really want to print "collection" after every protocol kind?

Reply:
I left in "collection" for `interpolated string collection` vs. (atomic) `string` because initially it wasn't obvious to me that an interpolated string was a collection. Should we assume that we won't need this differentiation for the average user and leave it out? `array` and `dictionary` definitely don't need "collection".

Reply:
I think `interpolated string` is good enough so you can drop "collection"